aml-linking.md

```diff
@@ -0,0 +1,153 @@
+* Defining `$base` without a `$id` will result in a **violation** in the document definition
+
+####  Example correct usage
+
+##### Dialect definition:
+```yaml
+MyNode:
+  mapping:
+    myLinkProperty:
+      range: schema.Person
+      allowMultiple: true
+      isLink: true
+```
+
+##### Document:
+```yaml
+myLinkProperty: 
+  - http://test.com/JohnDoe           # URI link string
+  - $id: http://test.com/JohnDoe      # URI link map 
+  - $id: http://another.com/JohnDoe   # URI link map with base
+    $base: http://test.com/
+```
+
+In this example all of the nodes in the array point to the same node using *different representations* of URI links.
+URI links can point to any node regardless of where it was defined, as long a URI is provided for it.   
+
+####  Example incorrect usage
+
+##### Dialect definition:
+```yaml
+MyNode:
+  mapping:
+    myLinkProperty:
+      range: schema.Person
+      allowMultiple: true
+      isLink: true
+```
+
+##### Document:
+```yaml
+myLinkProperty: 
+  - $id: http://test.com/JohnDoe      # Adds an extra property
+    extra: property 
+  - $base: http://test.com/           # Defines $base without $id
+```
+
+### ID template links
+ID template links use ID template definitions to construct links target URIs. This ID template definition comes from the 
+dialect responsible for parsing the target node, called the *target dialect*. The *target node mapping* is defined 
+consistently as the node mapping in the target dialect responsible for parsing the target node.
+
+> ID template links: link property value that represents the values for the variables defined in the id template of the 
+> target node mapping defined in the target dialect
+
+####  Example
+
+##### Target dialect definition:
+```yaml
+TargetNodeMapping:
+  idTemplate: http://test.com/{a}/{b}
+  mapping:
+    a:
+      range: string
+    b: 
+      renge: string
+```
+
+##### Target document:
+```yaml
+# Target node -> http://test.com/people/JohnDoe
+a: people
+b: JohnDoe
+```
+
+##### Source dialect definition:
+```yaml
+uses:
+  targetDialect: target-dialect.yaml
+nodeMappings:
+  MyNode:
+    mapping:
+      myLinkProperty:
+        range: targetDialect.TargetNodeMapping
+        isLink: true
+```
+
+##### Document:
+```yaml
+myLinkProperty: # Link to http://test.com/people/JohnDoe
+  a: people
+  b: JohnDoe
+```
+
+As you can see in the above example, both the target node and the link node use the same ID template definition to 
+construct the node ID & target ID respectively. For this it is necessary to import the target dialect with the `uses` 
+keyword and setting the target node mapping as the `range` of the link property. 
+
+
+[comment] Can we import node mappings from any dialect or do we need it to be a library?
+
+[comment] What happens when we mix URI links and ID template links (use $id in an idTemplate link)
+
+[comment] What is the expected behavior if the range node does not define an ID template. Can we fallback to URI links?
```

Comment:
Decis si esta marcado como is link? se puede usar un fallback o tirar una violation explicita. Si hacemos un fallback, al menos se podria dar un warning (vamos a procesar pero avisamos que esta raro)